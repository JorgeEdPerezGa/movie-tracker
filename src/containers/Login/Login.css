.page-title {
  color: #959595;
  padding-top: 40px;
  font-weight: 200;
  margin: 0;
  font-size: 40px; }

.login {
  background: #0c0c0c;
  min-height: 100vh; }

.login-form {
  padding: 30px;
  padding-top: 150px;
  padding-top: 40px;
  display: flex;
  justify-content: center; }

.login-input {
  font-size: 17px;
  margin-top: 34px;
  width: 400px;
  height: 30px;
  padding-left: 7px; }

.login-button {
  margin-bottom: 35px;
  background: #41aff8;
  border-radius: 100px;
  border: none;
  color: white;
  margin-top: 29px;
  height: 37px;
  width: 150px;
  font-size: 15px; }
  .login-button:hover {
    cursor: pointer;
    background: #e7eff7;
    color: #090e12; }
