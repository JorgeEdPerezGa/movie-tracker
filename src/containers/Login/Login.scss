.page-title {
  color: #959595;
  padding-top: 40px;
  font-weight: 200;
  margin: 0;
  font-size: 40px;
}

.login {
  background: #0c0c0c;
  min-height: 100vh;
}

.login-form {
  padding: 30px;
  padding-top: 150px;
  padding-top: 40px;
  display: flex;
  justify-content: center;
}

.login-input {
  font-size: 17px;
  margin-top: 34px;
  width: 400px;
  height: 30px;
  padding-left: 7px;
}

.login-button {
  margin-bottom: 35px;
  background: #41aff8;
  border-radius: 100px;
  border: none;
  color: white;
  margin-top: 29px;
  height: 37px;
  width: 150px;
  font-size: 15px;
  &:hover {
    cursor: pointer;
    background: #ffffff
    color: #090e12;
  }
  &:disabled {
    background: #7a8389;
    color: #c9c9c9;
    cursor: auto;
  }
}

.error-message {
  color: #fb7070;
  font-size: 20px;
  font-weight: 200;
}
