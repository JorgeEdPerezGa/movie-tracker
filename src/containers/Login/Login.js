import React, { Component } from 'react';
import './Login.css';

export class Login extends Component {
  constructor(){
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = (event) => {
    const {name, value} = event.target;
    this.setState({
      [name]: value
    });
  }

  render() {
    return (
      <div>
        <form>
          <input
            className='user-email'
            placeholder='email'
            name='email'
            type='text'
            value={this.state.email}
            onChange = {this.handleChange}/>
          <input
            className='user-password'
            placeholder='password'
            name='password'
            type='text'
            value={this.state.password}
            onChange = {this.handleChange}/>
          <button>ENTER</button>
        </form>
      </div>
    );
  }
}

export default Login;
