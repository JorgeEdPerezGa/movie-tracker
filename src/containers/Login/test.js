import React from 'react';
import { shallow } from 'enzyme';
import { Login } from './Login';

describe('login', () => {

  const renderedComponent = shallow(
    <Login />
  );

  it('test', () => {
    expect(true).toEqual(true);
  });

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should start with expected default state', () => {
    const expected = {email: '', password: '', 'error': false};
    
  });
});
