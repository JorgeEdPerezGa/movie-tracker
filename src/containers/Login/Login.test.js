import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Login from './Login';

describe('Login', () => {
  it.skip('should not pass', () => {
    expect(false).toEqual(true);
  })

  // pass down as props
  const mockHistory = {
    push: jest.fn();
  }
})