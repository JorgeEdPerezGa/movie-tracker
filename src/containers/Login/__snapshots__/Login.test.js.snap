// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    addAllFavorites={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
    loginUser={[Function]}
    updateMovies={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="page-title"
>
          WELCOME BACK
</h1>,
        <form
          className="login-form"
          onSubmit={[Function]}
>
          <div
                    className="input-container"
          >
                    <input
                              className="login-input"
                              name="email"
                              onChange={[Function]}
                              placeholder="email"
                              type="text"
                              value=""
                    />
                    <input
                              className="login-input"
                              name="password"
                              onChange={[Function]}
                              placeholder="password"
                              type="text"
                              value=""
                    />
                    <button
                              className="login-button"
                    >
                              ENTER
                    </button>
          </div>
</form>,
      ],
      "className": "login",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "WELCOME BACK",
          "className": "page-title",
        },
        "ref": null,
        "rendered": "WELCOME BACK",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="input-container"
>
            <input
                        className="login-input"
                        name="email"
                        onChange={[Function]}
                        placeholder="email"
                        type="text"
                        value=""
            />
            <input
                        className="login-input"
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        type="text"
                        value=""
            />
            <button
                        className="login-button"
            >
                        ENTER
            </button>
</div>,
          "className": "login-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="login-input"
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="text"
                value=""
/>,
              <input
                className="login-input"
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="text"
                value=""
/>,
              <button
                className="login-button"
>
                ENTER
</button>,
            ],
            "className": "input-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "login-input",
                "name": "email",
                "onChange": [Function],
                "placeholder": "email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "login-input",
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ENTER",
                "className": "login-button",
              },
              "ref": null,
              "rendered": "ENTER",
              "type": "button",
            },
          ],
          "type": "div",
        },
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="page-title"
>
            WELCOME BACK
</h1>,
          <form
            className="login-form"
            onSubmit={[Function]}
>
            <div
                        className="input-container"
            >
                        <input
                                    className="login-input"
                                    name="email"
                                    onChange={[Function]}
                                    placeholder="email"
                                    type="text"
                                    value=""
                        />
                        <input
                                    className="login-input"
                                    name="password"
                                    onChange={[Function]}
                                    placeholder="password"
                                    type="text"
                                    value=""
                        />
                        <button
                                    className="login-button"
                        >
                                    ENTER
                        </button>
            </div>
</form>,
        ],
        "className": "login",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "WELCOME BACK",
            "className": "page-title",
          },
          "ref": null,
          "rendered": "WELCOME BACK",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="input-container"
>
              <input
                            className="login-input"
                            name="email"
                            onChange={[Function]}
                            placeholder="email"
                            type="text"
                            value=""
              />
              <input
                            className="login-input"
                            name="password"
                            onChange={[Function]}
                            placeholder="password"
                            type="text"
                            value=""
              />
              <button
                            className="login-button"
              >
                            ENTER
              </button>
</div>,
            "className": "login-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="login-input"
                  name="email"
                  onChange={[Function]}
                  placeholder="email"
                  type="text"
                  value=""
/>,
                <input
                  className="login-input"
                  name="password"
                  onChange={[Function]}
                  placeholder="password"
                  type="text"
                  value=""
/>,
                <button
                  className="login-button"
>
                  ENTER
</button>,
              ],
              "className": "input-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "login-input",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "email",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "login-input",
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "password",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ENTER",
                  "className": "login-button",
                },
                "ref": null,
                "rendered": "ENTER",
                "type": "button",
              },
            ],
            "type": "div",
          },
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
